fastify-postgres/server.js

```diff
@@ -1,81 +1,90 @@
+"use strict";
+
 const fastify = require("fastify");
 
-const {
-  migration,
-  getAll,
-  getById,
-  create,
-  deleteById,
-  update,
-} = require("./query");
-const db = require("./db");
+const connectionString =
+  process.env.DATABASE_URL ||
+  "postgresql://postgres:postgres@localhost:5432/fastify_postgres?schema=public";
 
 function build(opts = {}) {
   const app = fastify(opts);
 
-  // You should'nt run your migration everytime your app boot
-  // make a script outsied of app that can be run before development
-  // or before deployment
-  migration();
+  app.register(require("fastify-postgres"), { connectionString });
 
   app.get("/", async (_request, reply) => {
     try {
-      const books = await getAll();
-      reply.send(books.rows);
+      const client = await app.pg.connect();
+      const { rows } = await client.query("SELECT * FROM books");
+      client.release();
```

Comment:
I would recommend the following:

```js
fastify.post('/user/:username', (req, reply) => {
  // will return a promise, fastify will send the result automatically
  return fastify.pg.transact(async client => {
    // will resolve to an id, or reject with an error
    const id = await client.query('INSERT INTO users(username) VALUES($1) RETURNING id', [req.params.username])

    // potentially do something with id

    return id
  })
})
```

and remove the try/catch. Note that in case of an error the client is never released

Reply:
Is it okay that the client never released when error happen?